lib/internal/url.js

```diff
@@ -219,7 +221,8 @@ class URL {
   constructor(input, base) {
     // toUSVString is not needed.
     input = '' + input;
-    if (base !== undefined && !(base instanceof URL))
+    this[urlInstance] = true;
```

Comment:
I might put the `urlInstance` in the prototype to avoid having to set this every single time an object is created.